### ISSUE TITLE
Usage of ssl.Purpose introduced Python 2.7.9+ requirement

### ISSUE THREAD

**Issue by bluenote10** [dask/distributed#1122]
Trying to run the latest devel branch on Ubuntu 14.04 (Python 2.7.6) fails with:

```
Traceback (most recent call last):
  File "/home/fabian/.virtualenvs/dask/bin/dask-scheduler", line 11, in <module>
    load_entry_point('distributed', 'console_scripts', 'dask-scheduler')()
  File "/home/fabian/github/distributed/distributed/cli/dask_scheduler.py", line 172, in go
    main()
  File "/home/fabian/.virtualenvs/dask/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/fabian/.virtualenvs/dask/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/fabian/.virtualenvs/dask/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/fabian/.virtualenvs/dask/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/fabian/github/distributed/distributed/cli/dask_scheduler.py", line 134, in main
    security=sec)
  File "/home/fabian/github/distributed/distributed/scheduler.py", line 209, in __init__
    self.connection_args = self.security.get_connection_args('scheduler')
  File "/home/fabian/github/distributed/distributed/security.py", line 126, in get_connection_args
    d['ssl_context'] = self._get_tls_context(tls, ssl.Purpose.SERVER_AUTH)
```

It looks like ea2405e541551b694b0fa095ef1da81d1cfb383e has introduced a dependence on newer Python 2.7 versions, because `ssl.Purpose` was only introduced in 2.7.9 (according to [docs](https://docs.python.org/2/library/ssl.html#ssl.create_default_context)). This breaks distributed's compatibility with older systems like Ubuntu 14.04 or Debian 7 (Python 2.7.3).

**Comment by pitrou** [comment]
Yes. We use not only that, but the whole `SSLContext` API which is new in 2.7.9. I don't intend to write specific fallback code for pre-2.7.9 Pythons, unless some client wants to pay for it, as there's a non-trivial development and maintenance cost.

As for older systems, our policy is generally to recommend using [conda](https://conda.io/docs/) (with or without [https://docs.continuum.io/anaconda](Anaconda)), which gets you up-to-date Python binaries for free, even on a very old Linux system.

**Comment by mrocklin** [comment]
How expensive is it to maintain compatibility with older versions of Python.  Not backport functionality but just keep the old non-ssl code running?

**Comment by mrocklin** [comment]
Ah, it looks like @bluenote10 may have provided a solution :)

**Comment by pitrou** [comment]
Good question. I had overlooked that this broke functioning even when not using TLS. I guess we can do that reasonably easily.

**Comment by bluenote10** [comment]
Yes, not supporting TLS for older versions would probably be fine. But it would be nice not to break it completely for older Python versions. Ubuntu 14.04 desktops and Debian 7 clusters are still fairly common.

**Comment by mrocklin** [comment]
Thank you for raising this issue and providing a fix @bluenote10 .  It's good that we caught this before a public release.

**Comment by pitrou** [comment]
Fixed by PR #1123. Thanks again!
